### PR TITLE
LA-192 Delete Shared Space

### PR DESCRIPTION
**Pull request by imGok**
deletework

### CODE REVIEW

File: domain/lib/src/usecases/shared_space/remove_shared_space_interactor.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  RemoveSharedSpaceInteractor(this._sharedSpaceRepository);
+
+  Future<Either<Failure, Success>> execute(
+    SharedSpaceId sharedSpaceId,
+  ) async {
+    try {
+      final sharedSpace = await _sharedSpaceRepository.deleteSharedSpace(sharedSpaceId);
+      return Right<Failure, Success>(DeleteSharedSpaceViewState(sharedSpace));
+    } catch (exception) {
```
Comment (hoangdat) [inline domain/lib/src/usecases/shared_space/remove_shared_space_interactor.dart:47]: how about the case `404` - shared space not found?

Comment (imGok) [reply domain/lib/src/usecases/shared_space/remove_shared_space_interactor.dart:47]: WDYM ? It's thrown by datasource and catch into this

Comment (hoangdat) [reply domain/lib/src/usecases/shared_space/remove_shared_space_interactor.dart:47]: I want for this case, We don't need to show toast for user, only reload the list

Comment (imGok) [reply domain/lib/src/usecases/shared_space/remove_shared_space_interactor.dart:47]: You mean if 404 is thrown => no toast ?

**Comment by imGok** [comment]
@manhduc-nguyen @hoangdat finished you can review

**Review by manhduc-nguyen — APPROVED** [review]

File: domain/lib/src/model/sharedspace/shared_space_role.dart [hunk 2]
```diff
@@ -40,6 +40,10 @@ class SharedSpaceRole extends Equatable {
 
   SharedSpaceRole(this.sharedSpaceRoleId, this.name);
 
+  factory SharedSpaceRole.initial() {
+    return SharedSpaceRole(SharedSpaceRoleId('unknown'), SharedSpaceRoleName.UNKNOWN);
```
Comment (hoangdat) [inline domain/lib/src/model/sharedspace/shared_space_role.dart:44]: why not use `Reader` instead?

File: domain/lib/src/usecases/shared_space/remove_shared_space_interactor.dart (hunk 1 above)
Comment (hoangdat) [reply domain/lib/src/usecases/shared_space/remove_shared_space_interactor.dart:47]: yes.

Comment (imGok) [reply domain/lib/src/usecases/shared_space/remove_shared_space_interactor.dart:47]: Done

**Review by hoangdat — APPROVED** [review]

**Pull request merged**